src/main/java/meteordevelopment/meteorclient/utils/world/BlockUtils.java

```diff
@@ -317,12 +333,31 @@ public enum MobSpawn {
 
     private static final ThreadLocal<BlockPos.Mutable> EXPOSED_POS = ThreadLocal.withInitial(BlockPos.Mutable::new);
 
-    public static boolean isExposed(BlockPos blockPos) {
-        for (Direction direction : Direction.values()) {
-            if (!mc.world.getBlockState(EXPOSED_POS.get().set(blockPos, direction)).isOpaque()) return true;
-        }
+    public static boolean isExposed(BlockPos blockPos, Integer distanceToCheck) {
+    	
+    	boolean isPosExposed = false;
+    	
+    	BlockState currBlockState = mc.world.getBlockState(blockPos);
+
+		outerloop:
+		for (Direction direction : Direction.values()) {
+			boolean foundAir = false;
+			for (int i = 1; i <= distanceToCheck+1; i++) {
+				if(i >= distanceToCheck && foundAir) {
+					isPosExposed = true;
+					break outerloop;
+				}
+				BlockState neighbourBlockState = mc.world.getBlockState(blockPos.offset(direction, i));
+				if(!mc.world.getBlockState(blockPos.offset(direction, i)).isOpaque()) {
+					foundAir = true;
+				} else {
+					break;
+				}
+            }
+		}
+    	
```

Comment:
Wtf even is this.
- The EXPOSED_POS was used to not create blockstate objects inside a hot loop. Your code does that and you didn't even clean up the EXPOSED_POS variable.
- you don't use the local variable neighbourBlockState which you just created
- you can return true early instead of setting isPosExposed = true and breaking the outer loop

- putting that return logic inside an if statement inside the inner loop is messy and you have a bug now. At least I think that [ore] [air] [block] should not be exposed if distance to check is 2.
```
	outerloop:
        for (Direction direction : Direction.values()) {
            for (int i = 1; i <= distanceToCheck; i++) {
                BlockState neighbourBlockState = mc.world.getBlockState(blockPos.offset(direction, i));
                if(neighbourBlockState.isOpaque()) {
                    continue outerloop;
                }
            }
            return true;
        }
```